src/Campaigns/resources/admin/components/CampaignFormModal/index.tsx

```diff
@@ -73,22 +41,70 @@ const removeTimezoneFromDateISOString = (date: string) => {
     return date.slice(0, -5);
 };
 
+/**
+ * Goal Type Option component
+ *
+ * @unreleased
+ */
+const GoalTypeOption = ({type, label, description, selected, register}: any) => {
+    const divRef = useRef(null);
+    const labelRef = useRef(null);
+
+    const handleDivClick = () => {
+        labelRef.current.click();
+    };
+
+    return (
+        <div
+            className={`${styles.goalTypeOption}  ${selected ? styles.goalTypeOptionSelected : ''}`}
+            ref={divRef}
+            onClick={handleDivClick}
+        >
+            <div className={styles.goalTypeOptionIcon}>
+                {type === 'amount' && <AmountIcon />}
```

Comment:
I like to define these things outside the component in a function that takes one parameter, in this case `type`, and use a `switch` statement to return the correct component.

Reply:
Resolved in 6c949216ae8cb5a970f4665d96f27abbab36fc5d